src/mqtt_client/mqtt_client.rs

```diff
@@ -166,6 +263,10 @@ where
 
             let response = MqttMessage::new(properties, default_response_topic, &message);
 
+            if rx_timeout.is_some() {
```

Comment:
We should check whether the message is retained (assuming we can). If the message is not retained, don't advance the timeout.

Reply:
Right now, I don't think minimq exposes the flags of the received message, just the topic, data, and properties, so we don't know if they are retained or not.

Reply:
Yeah. There is a race condition (retained timeout racing new settings coming in). But from the potential consequences it looks benign. The only general issue I see is that active settings updates may effectively DOS republication. I think that can only be solved by a dirty flag for each path (received a retained message) and an timeout since connection instead. Fine for now.